version.go

```diff
@@ -19,6 +19,10 @@ var (
 	ErrInvalidSemVer = errors.New("Invalid Semantic Version")
 )
 
+// Controls whether or not parsed constraints are cached
+var cacheVersions = true
```

Comment:
Why is caching enabled by default and cannot be turned off? Something other than Glide (that is short lived) like a server could easily have a cache that grows quite large causing memory issues. Any thoughts on handling this?